packages/code-studio/src/main/AppDashboards.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useCallback } from 'react';
+import classNames from 'classnames';
+import {
+  DashboardUtils,
+  DEFAULT_DASHBOARD_ID,
+  DehydratedDashboardPanelProps,
+  LazyDashboard,
+} from '@deephaven/dashboard';
+import { useConnection } from '@deephaven/jsapi-components';
+import { VariableDefinition } from '@deephaven/jsapi-types';
+import LayoutManager, { ItemConfigType } from '@deephaven/golden-layout';
+import { LoadingOverlay } from '@deephaven/components';
+import EmptyDashboard from './EmptyDashboard';
+
+interface AppDashboardsProps {
+  dashboards: {
+    id: string;
+    layoutConfig: ItemConfigType[];
+  }[];
+  activeDashboard: string;
+  onGoldenLayoutChange: (goldenLayout: LayoutManager) => void;
+  plugins: JSX.Element[];
+  onAutoFillClick: (event: React.MouseEvent) => void;
+}
+
+export function AppDashboards({
+  dashboards,
+  activeDashboard,
+  onGoldenLayoutChange,
+  plugins,
+  onAutoFillClick,
+}: AppDashboardsProps): JSX.Element {
+  const connection = useConnection();
+
+  const hydratePanel = useCallback(
```

Comment:
How will this work from Enterprise? We need to be able to pass in the hydration function and pull the query from the metadata for Enterprise.

Reply:
I don't think this hydration specifically will affect Enterprise. This is just moving the hydration function from `AppMainContainer` to `AppDashboards` since `AppMainContainer` doesn't need to know anything about this once the dashboards component is split out

Enterprise has its own hydration which would be part of the `Dashboard` already

As far as Enterprise changes go, it will need to add `dashboardPluginData` and also listen to the event for creating a dashboard like DHC does. There might be something w/ the plugin and plugin data that we need to modify to work w/ enterprise PQs since right now the plugin is responsible for fetching its  widget

Reply:
I've thought about extending `WidgetPlugin` in some manner so that it doesn't have to mount a panel. Right now it's kind of hacky IMO because the dashboard widget from dh.ui is loaded as part of a dashboard plugin. Instead it could be a widget plugin that mounts, but doesn't render a panel.

Or we need some sort of context/hook that lets the plugin get its own fetch function or create one with just the widget name and type. In DHE the hook could wrap any other info like PQ data into the fetch function it provides

Reply:
A plugin that mounts but doesn't have a panel sounds kinda neat. Need to give it more thought.